packages/typespec-ts/src/utils/namespaceUtils.ts

```diff
@@ -64,15 +67,37 @@ export function getOperationNamespaceInterfaceName(
 
 export function detectModelConflicts(dpgContext: SdkContext) {
   const allModels = getAllModels(dpgContext);
-  const nameSet = new Set<string>();
+  const nameSet = new Map<string, Model[]>();
+  let hasConflict = false;
   for (const model of allModels) {
     if (model.name === "") {
       continue;
     }
     if (nameSet.has(model.name)) {
-      return true;
+      nameSet.set(model.name, [
+        ...nameSet.get(model.name)!,
+        model.__raw! as Model
+      ]);
```

Comment:
nit: could we directly add items into original array? so that we don't need to build a new array every time.